bio_hansel/quality_check/__init__.py

```diff
@@ -0,0 +1,69 @@
+from typing import List, Callable, Tuple, Optional
+
+from pandas import DataFrame
+
+from bio_hansel.quality_check.quality_check_functions import check_missing_tiles, does_subtype_result_exist, \
+    check_mixed_subtype, check_intermediate_subtype, check_inconsistent_results
+from bio_hansel.quality_check.const import FAIL_MESSAGE, WARNING_MESSAGE
+from bio_hansel.subtype import Subtype
+import logging
+
+
+QC_FUNCS: List[Callable[[Subtype, DataFrame], Tuple[str, str]]] = \
+[
+    check_missing_tiles,
+    check_mixed_subtype,
+    check_inconsistent_results,
+    check_intermediate_subtype
+]
+
+
+'''
+[perform_quality_check]
+Input: Subtype, DataFrame
+Output: None, modifies the subtype with the result.
+Desc: This is the driver method for the quality check module. Every method within the QC_FUNCS list will be run 
+with parameters ( SUBTYPE, DATAFRAME ). If a quality check module returns something other than None, then an Error,
+or Warning has occured. 
+'''
+
+
+def perform_quality_check(st: Subtype, df: DataFrame):
+    logging.debug("Performing Quality Checking")
+    overall_qc_status = 'PASS'
+    messages = []
+
+    if does_subtype_result_exist(st) is False:
+        logging.warning("QC: Quality checking not run, subtype result did not exist.")
+        st.qc_status = 'FAIL'
+        st.qc_message = 'FAIL: Subtype does not exist, quality checking was not run.'
+        return None
+
+    for func in QC_FUNCS:
+        # Calls run_method to check that the qc function takes a Subtype, returns Tuple[Optional[str], Optional[str]]
+        status, message = run_method(func, st, df)
```

Comment:
Why not just call `func(st, df)` here?

Reply:
I think it was due to an earlier review comment where it suggested I check the type of the function to make sure it was a Callable[[Subtype, DataFrame], Tuple[str, str]]. Is there a better way to do this rather than having an extra method to check?

Reply:
I think you already have the checking going with specifying the type of the `QC_FUNCS` list:

```python
QC_FUNCS: List[Callable[[Subtype, DataFrame], Tuple[str, str]]] = \
[
    check_missing_tiles,
    check_mixed_subtype,
    check_inconsistent_results,
    check_intermediate_subtype
]
```

Reply:
Oh man, I put both in. I'll remove the run_method method.